### PR TITLE
DAOS-14262 cart: add ability to select traffic class for SWIM context 

### PR DESCRIPTION
**Pull request by soumagne**
### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by soumagne** [comment]
Will rebase PR once https://github.com/daos-stack/daos/pull/14662 has landed

File: src/cart/crt_hg.c [hunk 1]
```diff
@@ -904,10 +904,16 @@ crt_hg_class_init(crt_provider_t provider, int ctx_idx, bool primary, int iface_
...
-	init_info.request_post_incr = crt_gdata.cg_post_incr;
-
-	hg_class = HG_Init_opt(info_string, crt_is_service(), &init_info);
+	init_info.request_post_init         = crt_gdata.cg_post_init;
+	init_info.request_post_incr         = crt_gdata.cg_post_incr;
+	init_info.multi_recv_op_max         = crt_gdata.cg_mrecv_buf;
+	init_info.multi_recv_copy_threshold = crt_gdata.cg_mrecv_buf_copy;
+	/* With cxi provider, separate SWIM traffic in an effort to prevent potential congestion. */
+	if (provider == CRT_PROV_OFI_CXI && crt_is_service() &&
+	    crt_gdata.cg_auto_swim_disable == 0 && ctx_idx == crt_gdata.cg_swim_ctx_idx)
```
Comment (frostedcmos) [inline src/cart/crt_hg.c:913]: the check for cg_auto_swim_disable should not be done; it only tells you whether swim is to be enabled at startup, as opposed to being enabled manually after the context has been created (like some tests do). In both cases you would want to use a separate class.

Comment (soumagne) [reply src/cart/crt_hg.c:913]: OK I'll fix it after I rebase.

**Review by frostedcmos — DISMISSED** [review]

**Review by daltonbohning — COMMENTED** [review]
So far seems to be working as expected on aurora, using `SWIM_TRAFFIC_CLASS=low_latency`

**Review by jgmoore-or — APPROVED** [review]

**Review by frostedcmos — APPROVED** [review]

**Comment by soumagne** [comment]
memory alloc issue in hardware medium

**Comment by mchaarawi** [comment]
> memory alloc issue in hardware medium unrelated to PR

the issue is already fixed in latest master. please rebase.
the change do affect the tests that have failed, so we can't force land this.

**Pull request merged**